pkg/cluster/internal/create/nodes.go

```diff
@@ -82,85 +84,148 @@ func convertReplicas(nodes []config.Node) []config.Node {
 // that will host `kind` nodes
 func provisionNodes(
 	status *logutil.Status, cfg *config.Config, clusterName, clusterLabel string,
-) ([]nodes.Node, error) {
+) error {
 	defer status.End(false)
 
-	allNodes := []nodes.Node{}
+	_, err := createNodeContainers(status, cfg, clusterName, clusterLabel)
+	if err != nil {
+		return err
+	}
 
-	// convert replicas to normal nodes
-	// TODO(bentheelder): eliminate this when we have v1alpha3
-	configNodes := convertReplicas(cfg.Nodes)
-	// TODO(bentheelder): allow overriding defaultRoleOrder
-	sortNodes(configNodes, defaultRoleOrder)
+	status.End(true)
+	return nil
+}
 
-	// create a func(role string)(nodeName string)
-	nameNode := makeNodeNamer(clusterName)
+func createNodeContainers(
+	status *logutil.Status, cfg *config.Config, clusterName, clusterLabel string,
+) ([]nodes.Node, error) {
+	defer status.End(false)
 
-	// provision all nodes in the config
-	// TODO(bentheelder): handle implicit nodes as well
-	for _, configNode := range configNodes {
-		// name the node
-		name := nameNode(string(configNode.Role))
-
-		// create the node into a container (docker run, but it is paused, see createNode)
-		status.Start(fmt.Sprintf("[%s] Creating node container 📦", name))
-		var node *nodes.Node
-		var err error
-		// TODO(bentheelder): decouple from config objects further
-		switch string(configNode.Role) {
-		case constants.ExternalLoadBalancerNodeRoleValue:
-			node, err = nodes.CreateExternalLoadBalancerNode(name, configNode.Image, clusterLabel)
-		case constants.ControlPlaneNodeRoleValue:
-			node, err = nodes.CreateControlPlaneNode(name, configNode.Image, clusterLabel, configNode.ExtraMounts)
-		case constants.WorkerNodeRoleValue:
-			node, err = nodes.CreateWorkerNode(name, configNode.Image, clusterLabel, configNode.ExtraMounts)
-		}
-		if node != nil {
+	// create all of the node containers, concurrently
+	desiredNodes := nodesToCreate(cfg, clusterName)
+	status.Start("Preparing nodes " + strings.Repeat("📦", len(desiredNodes)))
+	nodeChan := make(chan *nodes.Node, len(desiredNodes))
+	errChan := make(chan error)
+	defer close(nodeChan)
+	defer close(errChan)
+	for _, desiredNode := range desiredNodes {
+		desiredNode := desiredNode // capture loop variable
+		go func() {
+			// create the node into a container (docker run, but it is paused, see createNode)
+			node, err := desiredNode.Create(clusterLabel)
+			if err != nil {
+				errChan <- err
```

Comment:
Missing returns here? Else we'll put multiple errors onto the chans

Reply:
yes this should return 👍

Reply:
returns added